ceno_zkvm/src/structs.rs

```diff
@@ -284,7 +288,17 @@ impl<E: ExtensionField> ZKVMWitnesses<E> {
         let mut combined_lk_mlt = vec![];
         let keys = self.lk_mlts.keys().cloned().collect_vec();
         for name in keys {
-            let lk_mlt = self.lk_mlts.remove(&name).unwrap().into_finalize_result();
+            let lk_mlt = if std::env::var("MOCK_PROVING").is_ok() {
```

Comment:
are we able to retain original logic and move this logic to mock_prover?
the goal is avoid introducing unnessesary debug condition check in production flow

Reply:
Unfortunately, this logic is called before invoking the mock prover, by that time `self.lk_mlts` is empty. So I don't think we can do that.

Reply:
Can't we change that?

Reply:
We can prevent `self.lk_mlts` from being emptied.

---

Original behavior: consume `self.lk_mlts` (regardless of `MOCK_PROVING` env)

This PR behaviour: do not consume `self.lk_mlts` if `MOCK_PROVING=true` but can consume it if `MOCK_PROVING=false`.

Alternative behavuour to remove debug env stuff from that file: do not consume `self.lk_mlts`. But this might affect performance.